modules/update-manifest-application.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-manifest-application_{context}"]
+= Understanding how manifests are applied during an update
+
+Some manifests supplied in a release image need to be applied in a certain order because of the dependencies between them.
+For example, the `CustomResourceDefinition` resource must be created before the matching custom resources.
+Additionally, there is a logical order in which the individual Cluster Operators need to be updated to minimize disruption in the cluster.
+The Cluster Version Operator implements this logical order through the concept of Runlevels.
+
+These dependencies are encoded in the filenames of the manifests in the release image:
+
+[source, terminal]
+----
+0000_<runlevel>_<component>_<manifest-name>.yaml
+----
+
+For example:
+
+[source, terminal]
+----
+0000_03_config-operator_01_proxy.crd.yaml
+----
+
+The CVO internally builds a dependency graph for the manifests, obeying the following rules:
+
+* During an update, manifests at a lower Runlevel will be applied before those at a higher Runlevel.
+
+* Within one Runlevel, manifests for different components may be applied in parallel.
+
+* Within one Runlevel, manifests for a single component will be applied in lexicographic order.
+
+The CVO then applies manifests following the generated dependency graph.
+
+For some resource types, the CVO monitors the resource after its manifest is applied, and only considers it to be successfully updated after the resource reaches a stable state.
+Achieving this state can take some time.
+This is especially true for Cluster Operators, which may perform their own update actions in the cluster after the CVO deploys their new versions.
+While the additional update actions take place, these Cluster Operators will temporarily set their `Progressing` condition to `True`.
+
+The CVO waits until all Cluster Operators in the Runlevel:
+
+* Are Available=True,
+
+* Are Degraded=False, and
```

Comment:
This [does not apply during the install phase][1].  And [maybe one day it won't apply at all][2].  Is it worth committing ourselves in formal docs, vs. handwaving about the CVO caring about _something_ in the ClusterOperator `status`?

This also tiptoes up against the current "Understanding how manifests are applied during an update" scoping.  You currently have "During an update, manifests at a lower Runlevel..." which acknowledges the presence of non-update CVO modes.  Does that mean we should acknowledge the install mode in this line?  Or should we consistently exclude all non-update modes?  Personally, I think folks often forget that the CVO is constantly reconciling the state of its managed manifests with the in-cluster resources, regardless of whether an update is going on.  And acknowledging that makes it more clear why `oc adm upgrade` and similar UIs can be used to check in on cluster state regardless of whether you're mid-update or not.  But it also takes some time to discuss the other modes, and if a customer does happen to be update-mode focused, the other-mode discussion will be distracting.  ⚖️🤷🔮

[1]: https://github.com/openshift/enhancements/blob/a0482b15dfb9338ebbd6e7c6d711302febf78b99/dev-guide/cluster-version-operator/user/reconciliation.md#clusteroperator
[2]: https://issues.redhat.com/browse/OTA-540

Reply:
>  Is it worth committing ourselves in formal docs, vs. handwaving about the CVO caring about something in the ClusterOperator status?

So in this case I'd argue for being explicit, because it helps understanding why exactly is CVO waiting on something (because CVO itself does not communicate it via our UIs, it jsut says "waiting in X-operator".

About the non-upgrade mode... You're making a really good point. In hindsight, maybe the doc could have been structured differently? We could have a section on "Cluster Version Operator is an operator that is active all the time and works like this", and then we could have a section that would explain how update is just an interest-worthy case of quite standard CVO operation. I'd like such doc! But I think we went a different route and in the current strucutre we may want to keep the focus on the upgrading mode.